Dockerfile.fakemachine-arch

```diff
@@ -1,11 +1,44 @@
+# Build unpackaged dependencies from source
+FROM archlinux:base-devel AS builder
+
+# Build user-mode linux from AUR
+ENV EUID=1
+WORKDIR /tmp
+RUN pacman -Syu --noconfirm base-devel \
+                            bc \
+                            git \
+                            inetutils \
+                            libffi && \
+    gpg --recv-keys 38DBBDC86092693E && \
+    git clone --depth=1 https://aur.archlinux.org/linux-usermode.git && \
+    cd linux-usermode && \
+    export MAKEFLAGS="-j$(nproc)" && \
+    makepkg -s
+
+# Build rust-libslirp from source
+RUN pacman -Syu --noconfirm rust \
+                            libslirp && \
+    git clone --depth=1 https://gitlab.freedesktop.org/slirp/libslirp-rs.git && \
+    cd libslirp-rs && \
+    cargo install --all-features --verbose --path . --root /usr/local
+
+# Build the main runtime image
 FROM archlinux:base-devel
 
 # Bits needed to run fakemachine
 RUN pacman -Syu --noconfirm qemu-headless \
                             busybox \
                             linux \
+                            libslirp \
                             --assume-installed initramfs
 
 # Bits needed to build fakemachine
 RUN pacman -Syu --noconfirm go \
                             git
+
+# Copy user-mode-linux binaries
+COPY --from=builder /tmp/linux-usermode/linux-usermode-*-x86_64.pkg.tar.zst /tmp/
+RUN pacman -U --noconfirm /tmp/linux-usermode-*-x86_64.pkg.tar.zst
```

Comment:
Here are some makepkg config tweaks from my system:
```
$ cat ~/.config/pacman/makepkg.conf
# Make CPU go brrr
MAKEFLAGS="-j$(nproc)"

# Don't compress, we're installing it straight away
PKGEXT=".pkg.tar"

# Include the debug symbols within the package, to aid debugging
OPTIONS+=(debug)
```

Reply:
Dunno if debug is worth it here, but added the no-compress one. Thanks!